File: examples/cle.basic/call_other_object_event/files/AddFunction.py
import starpy
Service = starpy._InitSimple("AddFunctionService","123",0,0);
Service._CreateAtomicSysRootItem("TestItem","");
AtomicClass,ErrorInfo = Service._CreateAtomicObjectSimple("TestItem","TestClass","","");
AtomicFunction,ErrorInfo = Service._CreateAtomicFunctionSimple(AtomicClass,"Add","VS_INT32 Add(VS_INT32 x,VS_INT32 y);","",False,False);
#define output event
AtomicOutEvent = Service._CreateAtomicOutEvent(AtomicClass,"OnTest","","",True);
Obj = Service._AtomicToObject(AtomicClass);
def Obj_Add(self,x,y) :  
    return x+y;
Obj.Add = Obj_Add;
# event 
def Obj_OnTest(self, Ev, Args,Type,Val) :
    print("Object Event is triggered");
    print(Ev["_SrcObject"]);
    print(Ev["_DesObject"]);
    print(Ev["_EventID"]);
    print(Ev["_ServiceGroupID"]);
    print(".......Arg..........");
    print( Args );
    print( Type );
    print( Val );
    return True,"Event Result",4567,888.88;
Obj.OnTest = Obj_OnTest;    

File: trunk/examples/cle.basic/call_other_object_event/files/AddFunction.lua
require "libstarcore"
Service = libstarcore._InitSimple("AddFunctionService","123",0,0);
AtomicClass,ErrorInfo = Service:_CreateAtomicObjectSimple("TestItem","TestClass","","");
AtomicFunction,ErrorInfo = Service:_CreateAtomicFunctionSimple(AtomicClass,"Add","VS_INT32 Add(VS_INT32 x,VS_INT32 y);","",false,false);
AtomicOutEvent = Service:_CreateAtomicOutEvent(AtomicClass,"OnTest","","",true);

Obj = Service:_AtomicToObject(AtomicClass);
function Obj:Add(x,y)
  return x+y;
end
function Obj:OnTest(Ev, Args,Type,Val)
	print("Object Event is triggered");
	print(Ev._SrcObject);
	print(Ev._DesObject);
	print(Ev._EventID);
	print(Ev._ServiceGroupID);
	print(".......Arg..........");
	print( Args );
	print( Type );
	print( Val );
    return true,"Event Result",4567,888.88;
end    
